tools/commands/integration_test.py

```diff
@@ -128,32 +126,136 @@ def run_integration_test(self, plugin_name, directory, timeout):
         else:
             # match.group(2) == 'Some tests failed.'
             return TestResult.fail(plugin_name, self.target_tuple, [
-                'flutter-tizen test integration_test failed, \
-                    see the output above for details.'
+                'flutter-tizen test integration_test failed, see the output above for details.'
             ])
 
 
+class EphemeralTarget(Target):
+    """A Tizen emulator that launches/poweroffs itself during test."""
+
+    def __init__(self, name, platform):
+        super().__init__(name, platform)
+
+    def run_integration_test(self, plugin_name, directory, timeout):
+        self.launch()
+        result = super().run_integration_test(plugin_name, directory, timeout)
+        self.power_off()
+        return result
+
+    def _find_id(self):
+        completed_process = subprocess.run('sdb devices',
+                                           shell=True,
+                                           cwd='.',
+                                           universal_newlines=True,
+                                           stderr=subprocess.PIPE,
+                                           stdout=subprocess.PIPE)
+        if completed_process.returncode != 0:
+            raise Exception('sdb failure.')
+
+        lines = completed_process.stdout.rstrip().split('\n')
+        for line in lines[1:]:
+            tokens = re.split('[\t ]', line)
+            id = tokens[0]
+            name = tokens[-1]
+            if name == self.name:
+                return id
+        raise Exception(f'Could not find connected target {self.name}')
+
+    def launch(self):
+        completed_process = subprocess.run(f'em-cli launch -n {self.name}',
+                                           shell=True,
+                                           cwd='.')
+        if completed_process.returncode != 0:
+            raise Exception(f'Target {self.name} launch failed.')
+        time.sleep(5)
```

Comment:
Is this `5 seconds` safe enough to guarantee that the launch is completed?

Reply:
As far as I know, there's no way to set callbacks or wait the emulator process so the number is set reasonably based on some testings. I haven't had any problems yet but if the emulator doesn't launch in that time, the program will emit the exception in line 160:
```python
raise Exception(f'Could not find connected target {self.name}')
```
I'll leave a comment about why the waiting is required.

Reply:
okay I understand. :)